### PR TITLE
Bug fixes for develop!

### LINKED ISSUE #150: Username replacement in Edits and threads is broken
i.e on slack edit a message and add a mention and it dosen't render correctly on matrix.

### LINKED ISSUE #151: Cross team matrix -> slack room substitutions dosen't work
We should only transform bridged rooms to slack channels if they are part of the same team as the channel we are sending the message to. Otherwise they render as `#unknown-channel` on slack and are therefore useless.

### PR DESCRIPTION
**Pull request by Cadair**
fixes #150 and also fixes #151 

Sorry this ended up being way more of an undertaking than I hoped.

Signed off by: Stuart Mumford (stuart@cadair.com)

### CODE REVIEW

**Comment by Cadair** [comment]
Have now tested the second commit, it appears to work.

File: lib/BaseSlackHandler.js [hunk 1]
```diff
@@ -48,15 +48,15 @@ BaseSlackHandler.prototype.getSlackRoomNameFromID = async function(id, token) {
     return id;
 };
 
-BaseSlackHandler.prototype.replaceChannelIdsWithNames = async function(message, token) {
+BaseSlackHandler.prototype.replaceChannelIdsWithNames = async function(message, text, token) {
```
Comment (Half-Shot) [inline lib/BaseSlackHandler.js]: is message still used? Also you are returning message but editing text?

File: lib/BaseSlackHandler.js [hunk 2]
```diff
@@ -48,15 +48,15 @@ BaseSlackHandler.prototype.getSlackRoomNameFromID = async function(id, token) {
...
     var main = this._main;
 
-    if (message.text === undefined) {
+    if (text === undefined) {
         return message;
     }
 
     // match all channelIds
-    var testForName = message.text.match(CHANNEL_ID_REGEX);
+    var testForName = text.match(CHANNEL_ID_REGEX);
```
Comment (Half-Shot) [inline lib/BaseSlackHandler.js]: [var](https://media.tenor.com/images/50c5ec11ca69d065be8715b9950f41eb/tenor.gif)

File: lib/BaseSlackHandler.js (hunk 1 above)
Comment (Cadair) [reply lib/BaseSlackHandler.js]: Yeah, message is being used to lookup teams and senders and stuff. It should be returning text.

File: lib/BaseSlackHandler.js (hunk 2 above)
Comment (Cadair) [reply lib/BaseSlackHandler.js]: Do you just want a bigger diff to update to ts? :p

File: lib/BaseSlackHandler.js [hunk 3]
```diff
@@ -48,15 +48,15 @@ BaseSlackHandler.prototype.getSlackRoomNameFromID = async function(id, token) {
     return id;
 };
 
-BaseSlackHandler.prototype.replaceChannelIdsWithNames = async function(message, token) {
+BaseSlackHandler.prototype.replaceChannelIdsWithNames = async function(message, text, token) {
     var main = this._main;
 
-    if (message.text === undefined) {
+    if (text === undefined) {
         return message;
```
Comment (Half-Shot) [inline lib/BaseSlackHandler.js]: should this be `text`?

File: lib/BaseSlackHandler.js (hunk 2 above)
Comment (Half-Shot) [reply lib/BaseSlackHandler.js]: *grumbles* okay fine, I already fixed this in TS

**Pull request merged**